30-remote-site-setup.sh

```diff
@@ -153,7 +158,18 @@ fi
 echo "Using Bosco tarball: $(bosco_findplatform --url)"
 for ruser in $users; do
     setup_ssh_config
-    [[ $cvmfs_wn_client == 'no' ]] && setup_endpoints_ini
+done
+
+###################
+# REMOTE COMMANDS #
+###################
+
+# We have to pick a user for SSH, may as well be the first one
+remote_os_info=$(fetch_remote_os_info "$(printf "%s\n" $users | head -n1)" "$remote_fqdn")
```

Comment:
Note that

```suggestion
remote_os_info=$(fetch_remote_os_info "$(echo "$users" | head -n1)" "$remote_fqdn")
```

would have worked just fine (quoting `"$users"` would preserve its embedded newlines).

But, closer to what you had originally, you could also just strip everything after the first newline without needing a subshell with:

```suggestion
remote_os_info=$(fetch_remote_os_info ${users%%$'\n'} "$remote_fqdn")
```

Reply:
I think Brian wanted to preserve newlines but also strip leading and trailing whitespace.

Reply:
Or, as i think you probably actually intended,

```suggestion
remote_os_info=$(fetch_remote_os_info "${users%%[[:space:]]*}" "$remote_fqdn")
```

(I think it's `:space:`, not `:blank:`; and for character classes you need `[[:class:]]`; otherwise you just get `:` and the individual letters between the `:`s)

Reply:
> I think Brian wanted to preserve newlines but also strip leading and trailing whitespace.

O_o
 
Well, as far as i can tell, all the examples just keep the first username in $users, which is apparently a newline-delimited list with no other whitespace:

```
users=$(cat /etc/grid-security/grid-mapfile /etc/grid-security/voms-mapfile | \
            awk '/^"[^"]+" +[a-zA-Z0-9\-\._]+$/ {print $NF}' | \
            sort -u)
```

that `print $NF` takes care of stripping whitespace - so the only separators are newlines.

Reply:
> (I think it's `:space:`, not `:blank:`; and for character classes you need `[[:class:]]`; otherwise you just get `:` and the individual letters between the `:`s)

Argh, yeah, that's what I was going for. 

> Well, as far as i can tell, all the examples just keep the first username in $users, which is apparently a newline-delimited list with no other whitespace:

Yup, exactly. I just wanted the first user from this list.

Reply:
As a side note, this kind of thing makes a good case for using arrays.